core/storage/user/gae_models.py

```diff
@@ -3011,3 +3011,141 @@ def get_export_policy(cls) -> Dict[str, base_models.EXPORT_POLICY]:
+        }
+
+    @staticmethod
+    def get_model_association_to_user(
+    ) -> base_models.MODEL_ASSOCIATION_TO_USER:
+        """Model is exported as one instance per user."""
+        return base_models.MODEL_ASSOCIATION_TO_USER.ONE_INSTANCE_PER_USER
+
+    @classmethod
+    def get_export_policy(cls) -> Dict[str, base_models.EXPORT_POLICY]:
+        """Model contains data to export corresponding to a user."""
+        return dict(super(cls, cls).get_export_policy(), **{
+            'invited_to_learner_groups_ids':
+                base_models.EXPORT_POLICY.EXPORTED,
+            'student_of_learner_groups_ids':
+                base_models.EXPORT_POLICY.EXPORTED,
+            'progress_sharing_permissions_list':
+                base_models.EXPORT_POLICY.EXPORTED
+        })
+
+    @classmethod
+    def delete_learner_group_references(cls, group_id: str) -> None:
+        """Delete all references of given learner group stored in learner group
+        users model.
+
+        Args:
+            group_id: str. The group_id denotes which group's data to delete.
+        """
+        found_models = cls.get_all().filter(
+            datastore_services.any_of(
+                cls.student_of_learner_groups_ids == group_id,
+                cls.invited_to_learner_groups_ids == group_id
+        ))
+
+        for learner_group_user_model in found_models:
+            # If the user has been invited to join the group, delete the
+            # group id from the invited_to_learner_groups_ids list.
+            if group_id in (
+                    learner_group_user_model.invited_to_learner_groups_ids):
+                learner_group_user_model.invited_to_learner_groups_ids.remove(
+                    group_id)
+                learner_group_user_model.update_timestamps()
+                learner_group_user_model.put()
+
+            # If the user is a student of the group, delete the group id
+            # from the student_of_learner_groups_ids list and the progress
+            # sharing permissions corresponding to that group.
+            elif group_id in (
+                    learner_group_user_model.student_of_learner_groups_ids):
+                learner_group_user_model.student_of_learner_groups_ids.remove(
+                    group_id)
+
+                updated_progress_sharing_permissions_list = []
+
+                for progress_sharing_permission in (
+                    learner_group_user_model
+                        .progress_sharing_permissions_list):
+                    if progress_sharing_permission['group_id'] != group_id:
+                        updated_progress_sharing_permissions_list.append(
+                            progress_sharing_permission)
+
+                learner_group_user_model.progress_sharing_permissions_list = (
+                    updated_progress_sharing_permissions_list)
+
+                learner_group_user_model.update_timestamps()
+                learner_group_user_model.put()
```

Comment:
Should this be done here? I think we usually only handle operations to this class of model itself and not to other models.

Reply:
Ah, right, my mistake. Then can we at least not `put` individually but do all puts at once?

Reply:
Hmm, yeah, it somehow slipped my mind while I was making other changes. Updated it. Thanks!